packages/render/src/ui/frame.base.tsx

```diff
@@ -0,0 +1,261 @@
+                        // track dimensions of the root if possible
+                        rootDimensionsRef.current =
+                          rootRef.current?.computeDimensions();
+
+                        // track aspect ratio of image
+                        previousFrameAspectRatioRef.current =
+                          frameUiState.frame.imageAspectRatio ?? "1.91:1";
+
+                        Promise.resolve(
+                          frameState.onButtonPress(
+                            // @todo change the type onButtonPress to accept partial frame as well because that can happen if partial frames are enabled
+                            frameUiState.frame as Frame,
+                            frameButton,
+                            index
+                          )
+                        ).catch((error) => {
+                          // eslint-disable-next-line no-console -- provide feedback to the user
+                          console.error(error);
+                        });
+                      },
+                    },
+                    theme.Button
+                  )
+                ),
+              },
+              theme.ButtonsContainer
+            ),
+      imageContainer: components.ImageContainer(
+        {
+          frameState: frameUiState,
+          aspectRatio:
+            (frameUiState.status === "loading"
+              ? previousFrameAspectRatioRef.current
+              : frameUiState.frame.imageAspectRatio) ?? "1.91:1",
+          image: components.Image(
+            frameUiState.status === "loading"
+              ? {
+                  status: "frame-loading",
+                  frameState: frameUiState,
+                  onImageLoadEnd,
+                  aspectRatio: previousFrameAspectRatioRef.current ?? "1.91:1",
+                }
+              : {
+                  status: "frame-loading-complete",
+                  frameState: frameUiState,
+                  src: frameUiState.debugImage ?? frameUiState.frame.image,
+                  aspectRatio: frameUiState.frame.imageAspectRatio ?? "1.91:1",
+                  onImageLoadEnd,
+                },
+            theme.Image
+          ),
+          messageTooltip:
+            currentFrameStackItem.status === "message" ||
+            currentFrameStackItem.status === "requestError"
+              ? components.MessageTooltip(
+                  {
+                    frameState: frameUiState,
+                    status:
+                      currentFrameStackItem.status === "requestError" ||
+                      currentFrameStackItem.type === "error"
+                        ? "error"
+                        : "message",
+                    message: getErrorMessageFromFramesStackItem(
+                      currentFrameStackItem
+                    ),
+                  },
+                  theme.MessageTooltip
+                )
+              : null,
+        },
+        theme.ImageContainer
+      ),
```

Comment:
keep in mind that the error tooltip should only show if it's a deliberate message from the frame server. i see this behaviour has changed to also show if an error is thrown on the client side e.g. onTransaction throwing `onTransaction did not return transaction id` - this should only show the toast, not the tooltip message in the debugger

Reply:
good catch, perhaps instead of handling error as message tooltip I could just provide an event handler and it is up to userland to actually implement that (by default `console.error`) ?

Reply:
that's what I thought. MessageTooltip will be used only for messages and if there is an error it is up to you how you handle that but it won't be handled by MessageTooltip. Only messages will be handled.

Reply:
sorry didn't read properly the first time. agreed!

Reply:
Ok now it calls `onError` in case of a request error.